### ISSUE TITLE
Linker error with MPLABX 5.35 and XC32 2.40

### ISSUE THREAD

**Issue by JoffreyCoffineau** [Microchip-MPLAB-Harmony/bootloader#4]
When trying to build the project "uart_bootloader_sam_e70_xult" in MPLABX 5.35 and XC32 2.40, I got this linker error:
```
c:/program files/microchip/xc32/v2.40/bin/bin/../../lib/gcc/pic32c/6.2.1/v7e-m/fpv5/hard/crti.o: Link Error: Could not allocate section .init, size = 4 bytes, attributes = code keep 
c:/program files/microchip/xc32/v2.40/bin/bin/../../lib/gcc/pic32c/6.2.1/v7e-m/fpv5/hard/crti.o: Link Error: Could not allocate section .fini, size = 4 bytes, attributes = code keep 
 Link Error: Could not allocate program memory
collect2.exe: error: ld returned 255 exit status
make[2]: *** [nbproject/Makefile-sam_e70_xult.mk:249: dist/sam_e70_xult/production/sam_e70_xult.X.production.hex] Error 255
make[1]: *** [nbproject/Makefile-sam_e70_xult.mk:91: .build-conf] Error 2
make: *** [nbproject/Makefile-impl.mk:39: .build-impl] Error 2
```

**Comment by amitraddi** [comment]
@JoffreyCoffineau  Thank you for reporting the issue.

 I am able to reproduce the issue with MPLABX v5.35 but not with MPLABX v5.30

MPLABX v5.35 does not have the option **"Do not link crt0 startup code"** in xc32-ld settings which is required by bootloader projects as it uses a custom linker script and startup file.
We have raised the issue to tools team and this option should be added back in next release of the IDE

For now the workaround would be to explicitly add **"-nostartfiles"** in **additional driver option** as shown in image
![image](https://user-images.githubusercontent.com/45988337/77627813-7697ce00-6f6d-11ea-8069-85e3ea64f2ad.png)

**Comment by amitraddi** [comment]
@JoffreyCoffineau  The issue is resolved with MPLABX 5.40. Can you please verify and close the issue
